src/gui/src/scriptWidget.cpp

```diff
@@ -226,32 +225,48 @@ void ScriptWidget::addTclResultToOutput(int return_code)
   }
 }
 
-void ScriptWidget::addBufferToOutput()
+void ScriptWidget::addLogToOutput(const QString& text, const QColor& color)
 {
-  // Show whatever we captured from the output channel in grey
-  for (auto& out : outputBuffer_) {
-    if (!out.isEmpty()) {
-      addToOutput(out, buffer_msg_);
-    }
-  }
-  outputBuffer_.clear();
+  addToOutput(text, color);
+}
+
+void ScriptWidget::addReportToOutput(const QString& text)
+{
+  addToOutput(text, buffer_msg_);
 }
 
 void ScriptWidget::addToOutput(const QString& text, const QColor& color)
 {
+  // disable updates while added text
+  output_->setUpdatesEnabled(false);
+
   // make sure cursor is at the end of the document
-  QTextCursor cursor = output_->textCursor();
-  cursor.movePosition(QTextCursor::End);
-  output_->setTextCursor(cursor);
+  output_->moveCursor(QTextCursor::End);
 
+  QString output_text = text;
+  if (text.endsWith('\n')) {
+    // remove last new line
+    output_text.chop(1);
+  }
+
+  // set new text color
   output_->setTextColor(color);
-  for (QString& text_line : text.split('\n')) {
+
+  QStringList output;
+  for (QString& text_line : output_text.split('\n')) {
+    // check for line length limits
     if (text_line.size() > max_output_line_length_) {
       text_line = text_line.left(max_output_line_length_-3);
       text_line += "...";
     }
-    output_->append(text_line);
+
+    output.append(text_line);
   }
+  // output new text
+  output_->append(output.join("\n"));
+
+  // reenable updates to display text
+  output_->setUpdatesEnabled(true);
```

Comment:
I only see one update to output_ so I'm not clear why enableUpdates is required.

Reply:
I'll check, but that is probably left over from when the `output_->append` was in the loop.